crates/router/src/connector/datatrans/transformers.rs

```diff
@@ -1,25 +1,27 @@
-use std::fmt::Debug;
+use std::{collections::HashMap, fmt::Debug};
 
-use common_utils::types::MinorUnit;
+use common_utils::{pii::Email, types::MinorUnit};
+use hyperswitch_domain_models::{
+    payment_method_data::Card,
+    router_flow_types::{Execute, RSync},
+};
+use hyperswitch_interfaces::consts;
 use masking::Secret;
 use serde::{Deserialize, Serialize};
 
 use crate::{
-    connector::{
-        utils as connector_utils,
-        utils::{CardData, PaymentsAuthorizeRequestData},
+    connector::utils::{
+        self as connector_utils, AddressDetailsData, CardData, PaymentsAuthorizeRequestData,
+        RouterData,
     },
     core::errors,
-    types::{
-        self,
-        api::{self, enums},
-        domain,
-        transformers::ForeignFrom,
-    },
+    services::{self},
+    types::{self, api::enums, domain, transformers::ForeignFrom},
 };
-
 const TRANSACTION_ALREADY_CANCELLED: &str = "transaction already canceled";
 const TRANSACTION_ALREADY_SETTLED: &str = "already settled";
+const REDIRECTION_SBX_URL: &str = "https://pay.sandbox.datatrans.com";
+const REDIRECTION_PROD_URL: &str = "https://pay.datatrans.com";
```

Comment:
Why have we hardcoded these URLs?

Reply:
This URL used in the 3DS redirection
```
    fn handle_response(
        &self,
        data: &types::PaymentsAuthorizeRouterData,
        event_builder: Option<&mut ConnectorEvent>,
        res: Response,
    ) -> CustomResult<types::PaymentsAuthorizeRouterData, errors::ConnectorError> {
        let response: datatrans::DatatransResponse = res
            .response
            .parse_struct("Datatrans PaymentsAuthorizeResponse")
            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
        event_builder.map(|i| i.set_response_body(&response));
        router_env::logger::info!(connector_response=?response);
        types::RouterData::try_from(types::ResponseRouterData {
            response,
            data: data.clone(),
            http_code: res.status_code,
        })
    }
```
in handle response there's no other way to retrieve the base/secondary_url (   connectors: &settings::Connectors, not there in handle_response)  ,Had to hard code  it